src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/EDateEOMonth.txt

```diff
@@ -0,0 +1,566 @@
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),-20000)
+Date(356,5,31)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),-24264)
+Date(1,1,31)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),-24265)
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),-24250)
+Date(2,3,31)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),1)
+Date(2023,2,28)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),2)
+Date(2023,3,31)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),5)
+Date(2023,6,30)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),6)
+Date(2023,7,31)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),10000)
+Date(2856,5,31)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),20000)
+Date(3689,9,30)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),95722)
+Date(9999,11,30)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),95723)
+Date(9999,12,31)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),95724)
+Error({Kind:ErrorKind.InvalidArgument})
+
+// 12/30/1899 is the starting date as it is the epoch, hence this returns 30 for the date
+// EXCEL difference: 12/31/9876 since the epoch is off by one day
+>> EDate(Blank(),95724)
+Date(9876,12,30)
+
+>> EDate(DateTime(2023,1,31,12,32,45,234),Blank())
+Date(2023,1,31)
+
+// 12/30/1899 is the starting date as it is the epoch, hence this returns 30 for the date
+// EXCEL difference: 1/0/1900 since the epoch is off by one day
+>> EDate(Blank(),Blank())
+Date(1899,12,30)
+
+// EDate with a Date only at thee end of the month
+// Ensure 31 is properly translated for other months with less than 31 days
+
+>> EDate(Date(2023,1,31),0)
+Date(2023,1,31)
+
+>> EDate(Date(2023,1,31),-1)
+Date(2022,12,31)
+
+>> EDate(Date(2023,1,31),-2)
+Date(2022,11,30)
+
+>> EDate(Date(2023,1,31),-6)
+Date(2022,7,31)
+
+>> EDate(Date(2023,1,31),-7)
+Date(2022,6,30)
+
+>> EDate(Date(2023,1,31),-1000)
+Date(1939,9,30)
+
+>> EDate(Date(2023,1,31),-10000)
+Date(1189,9,30)
+
+>> EDate(Date(2023,1,31),-100000)
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> EDate(Date(2023,1,31),-20000)
+Date(356,5,31)
+
+>> EDate(Date(2023,1,31),-24264)
+Date(1,1,31)
+
+>> EDate(Date(2023,1,31),-24265)
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> EDate(Date(2023,1,31),-24250)
+Date(2,3,31)
+
+>> EDate(Date(2023,1,31),1)
+Date(2023,2,28)
+
+>> EDate(Date(2023,1,31),2)
+Date(2023,3,31)
+
+>> EDate(Date(2023,1,31),5)
+Date(2023,6,30)
+
+>> EDate(Date(2023,1,31),6)
+Date(2023,7,31)
+
+>> EDate(Date(2023,1,31),10000)
+Date(2856,5,31)
+
+>> EDate(Date(2023,1,31),20000)
+Date(3689,9,30)
+
+>> EDate(Date(2023,1,31),95722)
+Date(9999,11,30)
+
+>> EDate(Date(2023,1,31),95723)
+Date(9999,12,31)
+
+>> EDate(Date(2023,1,31),95724)
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> EDate(Date(2023,1,31),Blank())
+Date(2023,1,31)
+
+// EOMonth with a DateTime at thee end of the month should give same results as EDate at the end of the month
+// Ensure 31 is properly translated for other months with less than 31 days
+
+>> EOMonth(DateTime(2023,1,31,12,32,45,234),0)
+Date(2023,1,31)
```

Comment:
Is the time part of the date truncated? It may be good to add a test which writes the value as a string to show the expected result:

    >> Text(EOMonth(DateTime(2023,1,31,12,34,56),0),"yyyy-mm-dd hh:mm:ss")

With the following result if we truncate it:

    "2023-01-31 00:00:00"

or this result if we don't:

    "2023-01-31 12:34:56"

Reply:
Good idea.  Value should also be an integer, added tests for that too.